staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/ratcheting.go

```diff
@@ -408,3 +382,87 @@ func (f ratchetingValueValidator) SetPath(path string) {
 func (f ratchetingValueValidator) Applies(source interface{}, valueKind reflect.Kind) bool {
 	return true
 }
+
+// Key returns the child of the reciever with the given name.
+// Returns nil if the given name is does not exist in the new object, or its
+// value is not correlatable to an old value.
+// If receiver is nil or if the new value is not an object/map, returns nil.
+func (l *CorrelatedObject) Key(field string) *CorrelatedObject {
```

Comment:
Do we need tests for these methods? (I'm looking commit-by-commit so sorry if that's done later).

Reply:
tests added